### PR TITLE
ENG-362: parameterize environment settings

### PR DESCRIPTION
**Pull request by actarsky**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,9 +1,12 @@
 locals {
   version = "0.21.5"
-  api_token_arn = "arn:aws:secretsmanager:us-west-2:143723790106:secret:customer/${var.customer_id}"
```
Comment (alanxo) [inline main.tf:3]: Shouldn't this also use var.xo_account_id?

File: main.tf [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
-  endpoint_url = "https://portal-api.xosphere.io/v1"
```
Comment (alanxo) [inline main.tf:4]: Should this also be a param that can be overwritten?

Comment (alanxo) [reply main.tf:4]: I see a var was added, in that case should this be removed?

File: main.tf [hunk 3]
```diff
@@ -654,14 +659,15 @@ data "aws_lambda_function" "terminator_lambda_function" {
 //instance-orchestrator
 resource "aws_lambda_function" "xosphere_instance_orchestrator_lambda_k8s_enabled" {
   count = length(var.k8s_vpc_security_group_ids) > 0  && length(var.k8s_vpc_subnet_ids) > 0 ? 1 : 0
-  s3_bucket = "xosphere-io-releases-${data.aws_region.current.name}"
+
```
Comment (alanxo) [inline main.tf:662]: Is this whitespace intentional?

File: main.tf [hunk 4]
```diff
@@ -691,14 +697,15 @@ resource "aws_lambda_function_event_invoke_config" "xosphere_instance_orchestrat
 
 resource "aws_lambda_function" "xosphere_instance_orchestrator_lambda" {
   count = length(var.k8s_vpc_security_group_ids) == 0  || length(var.k8s_vpc_subnet_ids) == 0 ? 1 : 0
-  s3_bucket = "xosphere-io-releases-${data.aws_region.current.name}"
+
```
Comment (alanxo) [inline main.tf:700]: Is this whitespace intentional?

**Review by alanxo — CHANGES_REQUESTED** [review]

**Comment by actarsky** [comment]
something bad happened with this pull request - opened #9 instead

**Pull request closed without merging**